tests/unit_tests/udfs/test_base_udf.py

```diff
@@ -96,7 +90,13 @@ def setup_tests_and_run(bucketfs_conn_name, bucketfs_conn, sub_dir, model_name):
     ("all given", "test_bucketfs_con_name", Connection(address=f"file:///test"),
      "test_subdir", "test_model")
 ])
-def test_model_downloader_all_parameters(description, bucketfs_conn_name, bucketfs_conn, sub_dir, model_name):
+@patch('exasol_transformers_extension.utils.bucketfs_operations.create_bucketfs_location_from_conn_object')
```

Comment:
To be honest, I am not a big fan of `@patch` for anything else as where you need to mock something you can control. And, in that case, we could control the bucketfs via the factory

Reply:
Yes, it would be my intention to go with how it is for now. But let me just take this as an opportunity to express my dislike of the current design that leans heavily on dependency injection. It is confusing. The code looks more complicated than it actually is. You wouldn't immediately realize that it's only there for testing. Then you have to keep doing a mental exercise of replacing the object returned by the factory with an instance of a class linked to the default value of the factory.